### PR TITLE
implement category estimates

### PR DESCRIPTION
**Pull request by mgross**
Feature that incorporates user estimates and priors for concentrations, sources, weights and target proxies by covariate group. (assigned @arunge  as you wanted to check before committing in master).

### CODE REVIEW

**Comment by arunge** [comment]
To summarize: 

Additional options for the tabs "Prior Info" (2) and "User Estimates" (3) were added if a particular model type (1) is selected and if covariates are enabled (4) :

image

image

image

image

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: ReSources
 Title: Food Reconstruction Using Isotopic Transferred Signals
-Version: 22.03.4
+Version: 22.03.5
```
Comment (arunge) [inline DESCRIPTION:3]:
```suggestion
Version: 22.04.01
```

Comment (arunge) [reply DESCRIPTION:3]: We now use calendar versioning

**Review by arunge — CHANGES_REQUESTED** [review]
Some general remarks:

- We now develope on the beta branch, not directly on the main. So please, in the future, open pull requests that aim to merge to the beta. For now, we can leave this and I will try to solve the  merge conflicts later, when taking the changes from the main into the beta.

- the checks do fail for oxcal calculation. I added a fresh model output to the test, but it did not help. For some reason some individuals are missing in the oxcal output: e.g. `individual_1, individual_3, individual_4`. Therefore the test fails.

```R
oxcalLines1
 [1] "Plot()"                                                                                           
 [2] "{"                                                                                                
 [3] "Curve(\"terrestrial\",\"IntCal20.14c\");"                                                         
 [4] "Curve(\"Marine20\",\"Marine20.14c\");"                                                            
 [5] "Delta_R(\"Aquatic1\",0,1);"                                                                       
 [6] "Curve(\"Marine20\",\"Marine20.14c\");"                                                            
 [7] "Delta_R(\"Aquatic2\",2,0.5);"                                                                     
 [8] ""                                                                                                 
 [9] "Mix_Curve(Individual_2,\"terrestrial\",\"Aquatic1\", 0.434,0.294); R_Date(“Individual_2“, NA,NA);"
[10] "Mix_Curve(Individual_2,\"terrestrial\",\"Aquatic1\", 0.144,0.155); R_Date(“Individual_2“, NA,NA);"
[11] "Mix_Curve(Individual_2,\"terrestrial\",\"Aquatic1\", 0.091,0.105); R_Date(“Individual_2“, NA,NA);"
[12] "Mix_Curve(Individual_2,\"terrestrial\",\"Aquatic1\", 0.11,0.128); R_Date(“Individual_2“, NA,NA);" 
[13] "Mix_Curve(Individual_2,\"terrestrial\",\"Aquatic1\", 0.231,0.188); R_Date(“Individual_2“, NA,NA);"
[14] "Mix_Curve(male,\"terrestrial\",\"Aquatic1\", 0.202,0.225); R_Date(“male“, NA,NA);"                
[15] "Mix_Curve(england,\"terrestrial\",\"Aquatic1\", 0.202,0.225); R_Date(“england“, NA,NA);"          
[16] "Mix_Curve(england.male,\"terrestrial\",\"Aquatic1\", 0.202,0.225); R_Date(“england.male“, NA,NA);"
[17] "Mix_Curve(Individual_5,\"Aquatic1\",\"Aquatic2\", 0.09,0.086);"                                   
[18] "Mix_Curve(Individual_5,\"Aquatic1\",\"Aquatic2\", 0.083,0.101);"                                  
[19] "Mix_Curve(Individual_5,\"Aquatic1\",\"Aquatic2\", 0.025,0.037);"                                  
[20] "Mix_Curve(Individual_5,\"Aquatic1\",\"Aquatic2\", 0.484,0.207);"                                  
[21] "Mix_Curve(Individual_5,\"Aquatic1\",\"Aquatic2\", 0.04,0.039);"                                   
[22] "Mix_Curve(female,\"Aquatic1\",\"Aquatic2\", 0.144,0.205);"                                        
[23] "Mix_Curve(germany,\"Aquatic1\",\"Aquatic2\", 0.144,0.205);"                                       
[24] "Mix_Curve(germany.female,\"Aquatic1\",\"Aquatic2\", 0.144,0.205);"                                
[25] ""                                                                                                 
[26] ""                                                                                                 
[27] "};"    
```

I use a filter on `parEstimates` to `"Source contributions_Carnivores"` in one test. There only remain values for `individual_2`. Is this the goal now?

image


@mgross Do you have an idea what could be the issue here? Did (the format of) the model output change?

**Comment by mgross** [comment]
@arunge I fixed that. The change increased complexity and I didnt't cover all cases in the first try.

**Review by arunge — APPROVED** [review]
@mgross Sorry for the delay, you can alway ping me on slack if it is urgent.
The tests now are ok, I also tested a model run. Looks good! You can merge now. :slightly_smiling_face:

**Pull request merged**